pkg/jobs/utils.go

```diff
@@ -12,23 +12,75 @@ package jobs
 
 import (
 	"context"
+	"fmt"
 
+	"github.com/cockroachdb/cockroach/pkg/clusterversion"
 	"github.com/cockroachdb/cockroach/pkg/jobs/jobspb"
 	"github.com/cockroachdb/cockroach/pkg/sql/isql"
 	"github.com/cockroachdb/cockroach/pkg/sql/sem/tree"
 	"github.com/cockroachdb/errors"
 )
 
-// RunningJobExists checks that whether there are any other jobs (matched by
-// payloadPredicate callback) in the pending, running, or paused status that
-// started earlier than the job with provided jobID.
-// If the provided jobID is a jobspb.InvalidJobID, this function checks if
-// exists any jobs that matches the payloadPredicate.
+// RunningJobExists checks that whether there are any job of the given types
+// in the pending, running, or paused status, optionally ignoring the job with
+// the ID specified by ignoreJobID as well as any jobs created after it, if
+// the passed ID is not InvalidJobID.
 func RunningJobExists(
 	ctx context.Context,
-	jobID jobspb.JobID,
+	ignoreJobID jobspb.JobID,
 	txn isql.Txn,
-	payloadPredicate func(payload *jobspb.Payload) bool,
+	cv clusterversion.Handle,
+	jobTypes ...jobspb.Type,
+) (exists bool, retErr error) {
+	if !cv.IsActive(ctx, clusterversion.V23_1BackfillTypeColumnInJobsTable) {
+		return legacyRunningJobExists(ctx, ignoreJobID, txn, jobTypes...)
+	}
+	if len(jobTypes) == 0 {
+		return false, errors.AssertionFailedf("must specify job types")
+	}
+
+	typeStrs := fmt.Sprintf("('%s'", jobTypes[0].String())
```

Comment:
you'd make me happier with a `strings.Builder` here.

Reply:
heh, I initially did that -- along with specializing the actually-called 1 and 2 arg cases to both be a single allocation -- but then told myself that two extra allocs here per call, once a minute, just didn't matter at least compared to actually running the query a few lines below. 

But if you had the same impulse, then that's two of us, so I put it back now.